### ISSUE TITLE
Add interactive modeling demonstrations (Linear Regression & Gradient Descent)

### ISSUE THREAD

**Issue by witusj** [hanbedrijfskunde/econan#3]
## Goal

Add interactive visualizations to help students understand data modeling concepts, specifically linear regression and gradient descent optimization.

## Interactive Demonstrations

Integrate these two interactive tools at the appropriate place in the curriculum:

1. **Linear Regression Demo**
   - URL: https://datadrivendecisions.github.io/regression/demo.html
   - Shows: How linear regression fits a line through data points
   - Learning outcome: Visual understanding of model fitting

2. **Gradient Descent Demo**
   - URL: https://datadrivendecisions.github.io/regression/gradientdescent.html
   - Shows: How the optimization algorithm finds the best fit
   - Learning outcome: Understanding iterative optimization process

## Suggested Placement

**Option 1: Week 5-6 Planning (Analysis & Modeling phase)**
- Location: `docs/weekplanning/index.html`
- Section: Week 5-6 "Data Preparation & Modeling" 
- Context: When students are doing CRISP-DM Modeling phase
- Add as a learning resource for both AI and Conventional paths

**Option 2: Materiaal Page (Cross-Path Materials)**
- Location: `docs/materiaal/index.html`
- Section: Cross-Path Materials (relevant for both AI and Conventional)
- Add a new card: "Data Modeling Concepts"

**Option 3: Both locations** (recommended)
- Brief mention + link in Week 5-6 planning
- Detailed explanation in materiaal page

### In weekplanning/index.html (Week 5-6):

Add to the activities list:

```html
<li>
    <strong>Modeling Concepts (optioneel verdieping)</strong><br>
    <span style="font-size: var(--font-size-sm); color: var(--color-gray-600);">
        Interactieve demos:
        <a href="https://datadrivendecisions.github.io/regression/demo.html" target="_blank">Linear Regression</a> |
        <a href="https://datadrivendecisions.github.io/regression/gradientdescent.html" target="_blank">Gradient Descent</a>
        <br>Begrijp hoe modeling algorithms werken onder de motorkap (beide paden)
    </span>
</li>
```

### In materiaal/index.html (Cross-Path Materials):

Add a new card in the Cross-Path Materials section:

```html
<div class="card">
    <div class="card-body">
        <h3 class="card-title">🎓 Data Modeling Concepts</h3>
        <p class="card-text">
            Interactieve visualisaties om te begrijpen hoe modeling algorithms werken.
            Relevant voor ZOWEL AI als Conventional path studenten.
        </p>
        <h4 style="margin-top: var(--space-4); font-weight: var(--font-weight-semibold);">
            Linear Regression
        </h4>
        <p style="font-size: var(--font-size-sm); margin-top: var(--space-2);">
            Zie hoe een regressielijn door data points wordt getrokken. 
            Speel met de data en zie direct hoe de lijn zich aanpast.
        </p>
        <a href="https://datadrivendecisions.github.io/regression/demo.html" 
           target="_blank" 
           class="btn btn-secondary" 
           style="margin-top: var(--space-3); width: 100%;">
            Open Linear Regression Demo →
        </a>
        
        <h4 style="margin-top: var(--space-6); font-weight: var(--font-weight-semibold);">
            Gradient Descent
        </h4>
        <p style="font-size: var(--font-size-sm); margin-top: var(--space-2);">
            Begrijp hoe het optimalisatie-algoritme stap voor stap de best-fit line vindt.
            Zie de "cost function" dalen terwijl het algoritme leert.
        </p>
        <a href="https://datadrivendecisions.github.io/regression/gradientdescent.html" 
           target="_blank" 
           class="btn btn-secondary" 
           style="margin-top: var(--space-3); width: 100%;">
            Open Gradient Descent Demo →
        </a>
        
        <div style="margin-top: var(--space-4); padding: var(--space-3); background-color: var(--color-primary-50); border-radius: var(--radius-md);">
            <p style="font-size: var(--font-size-sm); color: var(--color-primary-700);">
                💡 <strong>Waarom belangrijk?</strong><br>
                - AI Path: Begrijp wat er gebeurt "onder de motorkap" van AI models<br>
                - Conventional Path: Snap de wiskunde achter regressie in Python/R/Excel<br>
                - Beide paden: Kritisch kunnen evalueren of een model wel/niet past
            </p>
        </div>
    </div>
</div>
```

## Learning Outcomes

After engaging with these demos, students should be able to:
- **Explain** what linear regression does (finds best-fit line)
- **Understand** that "best fit" is defined by minimizing error
- **Recognize** that gradient descent is an iterative optimization process
- **Appreciate** why some models take longer to train than others
- **Critically evaluate** when linear models are/aren't appropriate

## Pedagogical Context

**Timing**: Week 5-6 (Modeling phase of CRISP-DM)

**Complexity Level**: 
- Foundation students: Just visual understanding
- Analytical students: Understand the math behind it
- Advanced students: Could implement from scratch in Python

**Cross-Path Relevance**:
- AI path students: Understand what AI does when you prompt "fit a regression model"
- Conventional path students: Understand the algorithm behind Python's sklearn.linear_model or R's lm()

## Alternative: Video Explanation

If interactive demos need context, could pair with:
- StatQuest video on Linear Regression
- 3Blue1Brown on Gradient Descent
- Grant Sanderson's visual explanations

## Implementation Priority

- **Phase 1** (Quick win): Add links in Week 5-6 planning
- **Phase 2** (Better UX): Add detailed card in materiaal page
- **Phase 3** (Optional): Create custom iframe embed with NL instructions

## Labels

- feature
- enhancement
- learning-resources
- modeling
